modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueArea.scala

```diff
@@ -1,17 +1,86 @@
 package edu.gemini.seqexec.web.client.components
 
+import diode.data.{Empty, Pot}
+import diode.react.ReactPot._
+import diode.react._
+import edu.gemini.seqexec.web.client.model.UpdatedQueue
 import edu.gemini.seqexec.web.client.semanticui.elements.icon.Icon
+import edu.gemini.seqexec.web.client.semanticui.elements.message.CloseableMessage
+import edu.gemini.seqexec.web.common.{SeqexecQueue, SequenceState}
 import japgolly.scalajs.react.vdom.prefix_<^._
 import japgolly.scalajs.react._
+
 import scalacss.ScalaCssReact._
 
+object QueueTableBody {
+  // Minimum rows to display, pad with empty rows if needed
+  val minRows = 5
+
+  def emptyRow(k: String) =
+    <.tr(
+      ^.key := k, // React requires unique keys
+      <.td("\u00a0"),
+      <.td("\u00a0"),
+      <.td("\u00a0"),
+      <.td(
+        SeqexecStyles.notInMobile,
+        "\u00a0")
+    )
+
+  val component = ReactComponentB[QueueArea.Props]("QueueTableBody")
+    .stateless
+    .render_P( p =>
+      <.tbody(
+        // Render after data arrives
+        p.queue().render( q =>
+          q.queue.map(Some.apply).padTo(minRows, None).collect {
+            case Some(s) =>
+              <.tr(
+                ^.classSet(
+                  "positive" -> (s.state == SequenceState.Running),
+                  "negative" -> (s.state == SequenceState.Error)
+                ),
+                ^.key := s"item.queue.${s.id}",
+                <.td(
+                  ^.cls := "collapsing",
+                  s.id
+                ),
+                <.td(s.state.toString),
+                <.td(s.instrument),
+                <.td(
+                  SeqexecStyles.notInMobile,
+                  s.error.map(_ => <.p(Icon("attention"), " Error")).getOrElse(<.p("-"))
+                )
+              )
+            case _ =>
+              emptyRow(s"time.queue." + (1000*math.random).toInt)
```

Comment:
Now that I look at this, the `random` part should be improved. Regardless, React requires that components in an array have a unique key so that it can efficiently render changes to the virtual dom. Interestingly they don't need to be globally unique, just in an array or sequence. in this case, I just took a shortcut but will improve it with a `zipWithIndex` call